plugin/pulp/plugin/cataloger.py

```diff
@@ -0,0 +1,33 @@
+class Cataloger(object):
+    """
+    Catalog content provided by content sources.
+    """
+
+    def __init__(self, config, url):
+        """
+        Initialize the Cataloger with a config and url.
+
+        :param config: The content source configuration.
+        :type config: dict
+
+        :param url: The URL for the content source.
+        :type url: str
+        """
+        self.config = config
+        self.url = url
```

Comment:
:+1:  for constructing the object with the configuration.  However, I think the `url` should be passed into `get_downloader()` instead because the configuration is part of the object's *state* and the `url` is not.  For example, the cataloger could used as:
```
cataloger = Cataloger(configuration)
for url in url_list:
   downloader = cataloger.get_downloader(url)
   ...
```